targets/TARGET_Realtek/TARGET_AMEBA/RTWInterface.cpp

```diff
@@ -75,7 +75,7 @@ static rtw_result_t scan_result_handler( rtw_scan_handler_result_t* malloced_sca
             ap.channel = record->channel;
             WiFiAccessPoint *accesspoint = new WiFiAccessPoint(ap);
             memcpy(&scan_handler->ap_details[scan_handler->ap_num], accesspoint, sizeof(WiFiAccessPoint));
-            delete[] accesspoint;
+            delete accesspoint;
         }
         scan_handler->ap_num++;
```

Comment:
Hi @0xc0170 I think in this part we use allocation is to borrow the data structure of this class so we might use this `scan_handler->ap_details[]` somewhere else as a pointer to an instance of `WiFiAccessPoint`. 
line 80` scan_handler->ap_num__;` seems like being mis-placed here, will fix this and do some tests. Thanks.